python/pyspark/streaming/dstream.py

```diff
@@ -333,33 +448,38 @@ def transformWith(self, func, other, keepSerializer=False):
         arguments of (`time`, `rdd_a`, `rdd_b`)
         """
         if func.__code__.co_argcount == 2:
-            oldfunc = func
+            oldfunc = cast(Callable[[RDD[T_co], RDD[U]], RDD[V]], func)
 
-            def func(_, a, b):
+            def func(_: datetime, a: RDD[T_co], b: RDD[U]) -> RDD[V]:
                 return oldfunc(a, b)
 
         assert func.__code__.co_argcount == 3, "func should take two or three arguments"
-        jfunc = TransformFunction(self._sc, func, self._jrdd_deserializer, other._jrdd_deserializer)
+        jfunc = TransformFunction(
+            self._sc,
+            func,
+            self._jrdd_deserializer,
+            other._jrdd_deserializer,
+        )
         dstream = self._sc._jvm.PythonTransformed2DStream(
-            self._jdstream.dstream(), other._jdstream.dstream(), jfunc
+            self._jdstream.dstream(), other._jdstream.dstream(), jfunc  # type: ignore[attr-defined]
```

Comment:
Shouldn't be necessary when  signature is fixed. Also, similar cases in the following methods.